addons/sale_project/models/project.py

```diff
@@ -338,7 +338,11 @@ def _get_sale_order_items_query(self, domain_per_model=None):
         )
 
         SaleOrderLine = self.env['sale.order.line']
-        sale_order_line_domain = [('order_id', 'any', [('analytic_account_id', 'in', self.analytic_account_id.ids)])]
+        sale_order_line_domain = [
+            '&',
+            ('order_id', 'any', [('project_id', '=', self.id)]),
```

Comment:
Hello @xavierbol ! 

Concerning your question:

It is true that if you apply the above flow, you will not link an analytic `account_id` to your project. However, if afterwards you do, this will not solve the issue. Imagine for instance that I install fsm industry afterwards. Then, an analytic account will be linked to my (already created) project and, but the issue will not disappear since the `analytic_account_id` of the sale order_line will not be recomputed by `_onchange_project_id` method as this one will not be re-triggered. In my commit message, I wanted to point out that setting an analytic account on the project after making the flow does not solve the issue for already created SO linked to the project 😊

About your concern, on the `project_id` I wanted to ask if you had an other suggestion perhaps ? Is there something else I could rely on ?

Reply:
TBH, the `project_id` is not really used in the codebase, I think. (we should certainly remove it in master, we will investigate) and so the issue that you describe will probably never happen.

In the common workflow, we will automatically generate an AA (= [A]nalytic [A]ccount) when the SO ([S]ales [O]rder) will be confirmed and generate the project and task according to the settings of the products selected inside the SO. 🙂 

If the user would like to set a project on its SO than we will also automatically set the AA of the project and that's it, just like a default value if I can say. 🙂

In resume, I don't think we have something to do for that issue. Let's just fix the initial issue reported by the customer. 🙂 

cc @aju-odoo

Reply:
Hello @xavierbol I have updated the diff, the commit and the tests accordingly 🙂 !